### PR TITLE
API groups e2e tests remove controllers

### LINKED ISSUE #3532: Flaky e2e tests
**What steps did you take and what happened:**
Velero E2E tests have non-deterministic waits/sleep that can cause the tests to occasionally produce false negatives.
We should use conditional waiting instead of non-deterministic waits to obtain more consistent test results.

**What did you expect to happen:**

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by codegold79**
# Summary of changes
- removed controllers and webhooks unnecessary for testing API group version feature
- replaced many of the sleeps with `--wait` and other more deterministic blocking methods

# Does your change fix a particular issue?
Fixes #3532 

# Please indicate you've done the following:
- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by codegold79** [comment]
@ashish-amarnath I have reason to believe a bug was introduced by Velero recently somewhere.
For my API Group Versions e2e tests, I created two images: current one (should be the same as main as of today, March 11, 2021) and an older one from when I first published my API Groups PR (published Feb 4, 2021). The older image consistently passes. The newest image always fails. I’m currently on the search to identify the bug.

**Comment by codegold79** [comment]
These tests currently do not pass because PR #3125 removed the API group version feature changes. PR #3571 bug issue was created to address the issue. In the meantime, in order to get tests to pass, build a local Velero version at a commit that is before the merge of PR #3125.

File: test/e2e/testdata/enable_api_group_versions/music_v2beta2_rockband.yaml [hunk 1]
```diff
@@ -5,9 +5,5 @@ metadata:
   annotations:
     rockbands.music.example.io/originalVersion: v2beta2
 spec:
-  # Add fields here
-  genre: '60s rock'
+  genre: "60s rock"
   numberComponents: 4
-  leadSinger: John
-  leadGuitar: George
-  drummer: Ringo
```
Comment (ashish-amarnath) [inline test/e2e/testdata/enable_api_group_versions/music_v2beta2_rockband.yaml:13]: was this removal intentional?

Comment (codegold79) [reply test/e2e/testdata/enable_api_group_versions/music_v2beta2_rockband.yaml:13]: Yes, these fields (leadSinger, leadGuitar, dummer) were only put there so that @brito-rafa's controller webhooks could set default values for API group version that didn't have those fields. We don't need them to test if a CRD and CR(s) were successfully backed up and restored.

**Comment by codegold79** [comment]
I put this PR in DRAFT status as it has merge conflicts.

**Comment by codegold79** [comment]
@dsu-igeek @zubron, this PR is again ready for review after I fixed merge conflicts introduced when PR #3726 was merged.

File: test/e2e/testdata/enable_api_group_versions/README.md [hunk 2]
```diff
@@ -4,18 +4,12 @@ This directory contains Kubernetes manifests that are used for the enable API gr
...
 
 ## Reference
 
-These manifests, listed below, come from two different sources: github.com/jetstack/cert-manager and github.com/brito-rafa/k8s-webhooks:
-
-cert-manager.yaml
-
-- source: https://github.com/jetstack/cert-manager/releases/download/v1.0.3/cert-manager.yaml
-- license: https://github.com/jetstack/cert-manager/blob/master/LICENSE
+These manifests, listed below, come from github.com/brito-rafa/k8s-webhooks:
```
Comment (dsu-igeek) [inline test/e2e/testdata/enable_api_group_versions/README.md:12]: Can we move these into the Velero github somewhere or into a non-personal repo?

**Review by dsu-igeek — COMMENTED** [review]
Just a couple of small changes, thanks!

File: test/e2e/enable_api_group_versions_test.go [hunk 3]
```diff
@@ -54,29 +51,43 @@ var _ = Describe("[APIGroup] Velero tests with various CRD API group versions",
...
+				"EnableAPIGroupVersions", // TODO: remove when feature flag is removed
+			)
+			Expect(err).NotTo(HaveOccurred())
+		}
 	})
 
 	AfterEach(func() {
-		cmd := exec.CommandContext(ctx, "kubectl", "delete", "namespace", "music-system")
+		cmd := exec.CommandContext(ctx, "kubectl", "delete", "crd", resource+"."+group)
 		_, _, _ = veleroexec.RunCommand(cmd)
```
[the remark this answers is not in the document]
Comment (codegold79) [reply test/e2e/enable_api_group_versions_test.go:84]: Will do.

File: test/e2e/testdata/enable_api_group_versions/README.md (hunk 2 above)
Comment (codegold79) [reply test/e2e/testdata/enable_api_group_versions/README.md:12]: I'm not sure I understand what you're asking. The manifests (listed in lines 14-56) are in the current PR. I only mentioned @brito-rafa's repo to give credit to him since I copied/pasted his manifest files into this PR.

**Comment by codegold79** [comment]
Looks like merge conflicts were introduced with PR #3764 less than a week ago. I'll have to address these.

**Review by carlisia — COMMENTED** [review]
@codegold79 the test failed, please see the output: https://snippets.cacher.io/snippet/07d40f30c84fa9f49bcd

**Review by carlisia — COMMENTED** [review]
This PR lgtm, it's basically a removal of a bunch of resources.  The failure I got was about generating the data, maybe a timeout. 


Next: I'm going to run what we have on the `main` branch and see if that passes.

**Comment by codegold79** [comment]
> This PR lgtm, it's basically a removal of a bunch of resources. The failure I got was about generating the data, maybe a timeout.
> 
> Next: I'm going to run what we have on the `main` branch and see if that passes.

Thanks @carlisia for running the tests. I'll stand by and see if I can make any changes to help get the tests pass.

**Comment by carlisia** [comment]
@codegold79 it got worse: https://snippets.cacher.io/snippet/137ff753cef67263363a

**Comment by codegold79** [comment]
> @codegold79 it got worse: https://snippets.cacher.io/snippet/137ff753cef67263363a

Oh no! Kibishii tests are really struggling :(

**Review by carlisia — APPROVED** [review]
These tests run successfully on AWS.

All the chocking reported previously was mostly due to race conditions from things running in the same namespace. I fixed those and will open a PR tomorrow. No changes were made to this test after the race conditions and they pass. successfully

**Review by dsu-igeek — APPROVED** [review]
LGTM
